services/common/auth_tools.py

```diff
@@ -0,0 +1,369 @@
+
+
+class EnvironWithoutOrg:
+    def __init__(self, user_info: UserInfo):
+        self.user_info = user_info
+        self.organization = None
+        self.role = None
+
+
+class EnvironWithOrg(EnvironWithoutOrg):
+    def __init__(self, user_info: UserInfo, organization: str, role: ORG_ROLE_LITERAL):
+        self.user_info = user_info
+        self.organization = organization
+        self.role = role
+
+
+####################################### Restrictions By Organization #######################################
+def get_rsu_dict_for_org(organizations: list[str]) -> dict[str, str]:
+    if not organizations:
+        return {}
+    allowed_orgs_str = ", ".join(f"'{org}'" for org in organizations)
+    query = (
+        "SELECT rsu.ipv4_address::text AS ipv4_address "
+        "FROM public.rsus rsu "
+        "JOIN public.rsu_organization AS rsu_org ON rsu_org.rsu_id = rsu.rsu_id "
+        "JOIN public.organizations AS org ON org.organization_id = rsu_org.organization_id "
+        f"WHERE org.name = ANY (ARRAY[{allowed_orgs_str}])"
+    )
+
+    logging.debug(f'Executing query: "{query};"')
+    data = pgquery.query_db(query)
+
+    return {rsu["ipv4_address"]: rsu["ipv4_address"] for rsu in data}
+
+
+def check_rsu_with_org(rsu_ip: str, organizations: list[str]) -> bool:
+    if not organizations:
+        return {}
+    allowed_orgs_str = ", ".join(f"'{org}'" for org in organizations)
+    query = (
+        "SELECT rsu.ipv4_address::text AS ipv4_address "
+        "FROM public.rsus rsu "
+        "JOIN public.rsu_organization AS rsu_org ON rsu_org.rsu_id = rsu.rsu_id "
+        "JOIN public.organizations AS org ON org.organization_id = rsu_org.organization_id "
+        f"WHERE org.name = ANY (ARRAY[{allowed_orgs_str}]) "
+        f"AND rsu.ipv4_address = '{rsu_ip}'"
+    )
+
+    logging.debug(f'Executing query: "{query};"')
+    data = pgquery.query_db(query)
+
+    return data[0]["ipv4_address"] == rsu_ip if data else False
+
+
+def check_intersection_with_org(intersection_id: str, organizations: list[str]) -> bool:
+    if not organizations:
+        return {}
+    allowed_orgs_str = ", ".join(f"'{org}'" for org in organizations)
+    query = (
+        "SELECT intersection.intersection_number as intersection_number "
+        "FROM public.intersections rsu "
+        "JOIN public.intersection_organization AS intersection_org ON intersection_org.intersection_id = intersection.intersection_id "
+        "JOIN public.organizations AS org ON org.organization_id = intersection_org.organization_id "
+        f"WHERE org.name = ANY (ARRAY[{allowed_orgs_str}])"
+        f"AND intersection.intersection_number = '{intersection_id}'"
+    )
+
+    logging.debug(f'Executing query: "{query};"')
+    data = pgquery.query_db(query)
+
+    return data[0]["intersection_number"] == intersection_id if data else False
+
+
+def check_user_with_org(user_email: str, organizations: list[str]) -> bool:
+    if not organizations:
+        return {}
```

Comment:
chore(blocking): the return type of the method is bool, so we should return a bool not an empty dict
```suggestion
        return False
```